dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -132,11 +138,16 @@ private LimitedChannel createLimitedChannel(String uri, int uriIndex) {
     }
 
     private static LimitedChannel getUpdatedNodeSelectionStrategy(
-            @Nullable LimitedChannel previousNodeSelectionStrategy,
+            LimitedChannel previousNodeSelectionStrategy,
             ClientConfiguration config,
             List<LimitedChannel> channels,
             Random random,
             String channelName) {
+        if (channels.isEmpty()) {
+            log.info("Zero URIs available, all requests will be queued", SafeArg.of("channelName", channelName));
+            return ZeroUriChannel.INSTANCE;
```

Comment:
In this case I think we want to fail all requests due to a non-existant server -- draining the queue. We should make sure that the failure doesn't produce retries.

Reply:
Hmm, I kinda liked the idea of queuing up requests when we're in a zero uri state just in case there are blips in service discovery?

Reply:
I think that would be confusing for consumers, WC already uses channels that fail immediately when zero URIs are present.
Waiting for a URI to become available could be reasonable if we had a timeout, and possibly updated the queue logic to rotate requests so that a single request isn't queued forever while subsequent requests past the queue limit are always rejected.